src/tests/hdf5/test_generator.py

```diff
@@ -65,9 +65,34 @@ def test_hd5_data_generator_multiple_tensors(hd5_file):
     assert index == data_length - 1
 
 
-def test_hd5_with_label(hd5_file):
+def test_hd5_with_1d_label(hd5_file):
     """
-    Test for yielding labels alongside data.
+    Test for yielding labels alongside data, 1D label.
+    """
+    hd5_file_path, data_length = hd5_file
+    # GIVEN a HD5 data generator reading from HD5 file
+    hd5_data_generator = Hd5DataGenerator(hd5_file_path=hd5_file_path,
+                                          output_tensor_format=[['dataset0', 'dataset1'], ['dataset2']],
+                                          label='label',
+                                          expand_1d_categorical_to_2d=False)
+    # WHEN iterating over data
+    data_iter = hd5_data_generator()
+    for i, xy in enumerate(data_iter):
```

Comment:
```suggestion
    for i, xy in enumerate(data_iter):
```
renaming i, xy into something more readable helps reviewers understand what is supposed to happen a bit more.